examples/snap-plugin-collector-rand/rand/rand.go

```diff
@@ -117,8 +138,15 @@ func (RandCollector) CollectMetrics(mts []plugin.Metric) ([]plugin.Metric, error
 	The metrics returned will be advertised to users who list all the metrics and will become targetable by tasks.
 */
 func (RandCollector) GetMetricTypes(cfg plugin.Config) ([]plugin.Metric, error) {
-	metrics := []plugin.Metric{}
+	if req && len(cfg) < 1 {
```

Comment:
When `req` is true, it adds a new string rule where a config is required to be passed in. This is mocking the behavior of a plugin that requires a config in order to load. So on this line we are checking the case when `req` is true, did they pass in a config? If they did not provide a config `if len(cfg) < 1` then an error is thrown. So, this section is necessary.